pixeltable/tests/test_dataframe.py

```diff
@@ -295,18 +296,26 @@ def restrict_json_for_default_collate(obj):
         # test we get all rows
         def check_recover_all_rows(ds, size : int, **kwargs):
             dl = torch.utils.data.DataLoader(ds, **kwargs)
-            loaded_ids = set()
+            loaded_ids = []
             for batch in dl:
                 for row_id in batch['row_id']:
                     val = int(row_id) # np.int -> int or will fail set equality test below.
                     assert val not in loaded_ids, val
-                    loaded_ids.add(val)
+                    loaded_ids.append(val)
 
-            assert loaded_ids == set(range(size))
+            assert set(loaded_ids) == set(range(size))
+            return loaded_ids
+
+        loaded_ids = check_recover_all_rows(ds, size=df_size, batch_size=3, num_workers=0) # within this process
+        assert loaded_ids != list(range(df_size)) # check ordering is not preserved
+
+        ds_notshuffled = df.to_pytorch_dataset(image_format='pt', shuffle=False)
+        loaded_ids = check_recover_all_rows(ds_notshuffled, size=df_size, batch_size=3, num_workers=0)
+        assert loaded_ids == list(range(df_size)) # check ordering is preserved
```

Comment:
This isn't guaranteed to work, because the ordering might change, depending on what Postgres decides. In this case, you should supply an explicit order_by().